runtime/calamari/src/fee.rs

```diff
@@ -67,3 +47,104 @@ impl WeightToFeePolynomial for WeightToFee {
 		}]
 	}
 }
+
+#[cfg(test)]
+mod multiplier_tests {
+	use crate::{Runtime, RuntimeBlockWeights as BlockWeights, System, TransactionPayment, KMA};
+	use frame_support::weights::{DispatchClass, Weight, WeightToFeePolynomial};
+	use pallet_transaction_payment::{Multiplier, TargetedFeeAdjustment};
+	use polkadot_runtime_common::{AdjustmentVariable, MinimumMultiplier, TargetBlockFullness};
+	use sp_runtime::{
+		traits::{Convert, One},
+		FixedPointNumber,
+	};
+
+	fn run_with_system_weight<F>(w: Weight, assertions: F)
+	where
+		F: Fn() -> (),
+	{
+		let mut t: sp_io::TestExternalities = frame_system::GenesisConfig::default()
+			.build_storage::<Runtime>()
+			.unwrap()
+			.into();
+		t.execute_with(|| {
+			System::set_block_consumed_resources(w, 0);
+			assertions()
+		});
+	}
+
+	// update based on runtime impl.
+	fn runtime_multiplier_update(fm: Multiplier) -> Multiplier {
+		TargetedFeeAdjustment::<
+			Runtime,
+			TargetBlockFullness,
+			AdjustmentVariable,
+			MinimumMultiplier,
+		>::convert(fm)
+	}
+
+	// Consider the daily cost to fully congest our network to be defined as:
+	// `target_daily_congestion_cost_usd = inclusion_fee * blocks_per_day * kma_price`
+	// Where:
+	// `inclusion_fee = fee_adjustment * (weight_to_fee_coeff * (block_weight ^ degree)) + base_fee + length_fee`
+	// Where:
+	// `fee_adjustment` and `weight_to_fee_coeff` are configurable in a runtime via `FeeMultiplierUpdate` and `WeightToFee`
+	// `fee_adjustment` is also variable depending on previous block's fullness
+	// We are also assuming `length_fee` is negligible for small TXs like a remark or a transfer.
+	// This test loops 1 day of parachain blocks (7200) and calculates accumulated fee if every block is almost full
+	#[test]
+	fn congested_chain_simulation() {
+		// Configure these values depending on current prices and target cost.
+		let kma_price = 0.012f32;
+		let target_daily_congestion_cost_usd = 250000;
```

Comment:
I would propose to scale the target usd cost with kma price like so:

```
let current_kma_price = 0.012f32
let orig_kma_price = 0.012f32;
let target_daily_congestion_cost_usd = 250000 * (math::round::ceil(current_kma_price,current_kma_price / 0.012f32) as u32) ;
```

That way the USD price target doesn't have to be manually adjusted to reflect the (growing) economic value of the network.
At current marketcap the 250k target equals 0.2% of the network - and assuming KMA price is an accurate reflection of network value, it will stay at 0.2% that way.

( current price ofc still has to be manually adjusted )

Reply:
Yeah that's cool.

Reply:
std::cmp::max was what i intended to use there

Reply:
the coingecko API can fetch KMA price using below REST call 
https://api.coingecko.com/api/v3/simple/price?ids=calamari-network&vs_currencies=usd
and returns
```
{
  "calamari-network": {
    "usd": 0.01092173
  }
}
```
I'll try to whip up a call integrated into this test using reqwest